hash_table/[M]923.cpp

```diff
@@ -0,0 +1,33 @@
+// 923. 3Sum With Multiplicity - Medium
+// https://leetcode.com/problems/3sum-with-multiplicity/
+
+/* Analysis:
+- We maintain a hash table to save all numbers in the array.
+- We use 2 nested for loops to find all possible pair j, k, meanwhile all previous number i are already in the hastable.
+- We calculate the sum and find the number of i in the hash table and add to result.
+- After that, we save j into hash table and go next.
+- Time complexity: O(n^2)
+- Space complexity: O(n)
+*/
+
+class Solution {
+public:
+    int threeSumMulti(vector<int>& arr, int target) {
```

Comment:
A better solution in term of space complexity can be sorting the array and using 2 pointers.